p256/fiat_p256.ml

```diff
@@ -36,9 +36,13 @@ let%expect_test "dh" =
 
 type point = Point.t
 
-let point_of_hex = Point.of_hex
+let check_point = function
+  | Some p when not (Point.is_infinity p) -> Some p
+  | _ -> None
```

Comment:
This interface seems to lose information about why a number is rejected. Would it be better to encode it as a Result type so that the Error case can specifically say "infinity rejected" here?

Reply:
Right, this is done in #26, but these two PRs interfere for now ; I guess we'll add the infinity case to #26 after this PR is merged, if that sounds good to you.

Reply:
Ha indeed! Ordering of PRs entirely up to you; I'm just reading them out of order :-)